Make support for unnamed extensions more clear and explicit (#4616)
dart-lang/site-www: src/_guides/language/extension-methods.md

```
<div class="prettify-filename">lib/string_apis.dart</div>

The members of an extension can be methods, getters, setters, or operators.
Extensions can also have static fields and static helper methods.

### Unnamed extensions

You can optionally omit the name when declaring an extension.
An unnamed extension is visible only
in the library where it's declared.
You also can't apply it explicitly

> to resolve [API conflicts](#api-conflicts).

Author: Thanks for feedback. I've decided to rewrite the API conflict sentence to provide some extra context.

I needed to keep the word "explicitly" since its a specific type of usage mentioned above, in explicitly using the name to apply an extension rather than applying it directly through the underlying type. 

Let me know what you think of the adjusted text.

<?code-excerpt "extension_methods/lib/string_extensions/string_apis_unnamed.dart (unnamed)"?>
```dart
extension on String {
  bool get isBlank => trim().isEmpty;
}
```

> Note that without a name,
> an extension's static members can only be invoked
> within the extension declaration.

Reviewer: **nit:** Starting a sentence with note should either not be done or included in a note callout. If this doesn't rise to that occasion, exclude it.
```suggestion
Without a name, you can invoke an extension's
static members only within the extension declaration.
```

**question:** Is this not redundant with what's written in [lines 221 to 225](https://github.com/dart-lang/site-www/pull/4616/files#diff-3d07c88e0a0b6c36d93bad9dccd5211a89f70f70f94bedd205a61413a6b932d1R221-R225)?

Author: As for its relation to L221-225, it's a different class of limitation that is less likely to be come across.

```
<div class="prettify-filename">lib/string_apis.dart</div>

The members of an extension can be methods, getters, setters, or operators.
Extensions can also have static fields and static helper methods.

### Unnamed extensions

> When declaring an extension, you can omit the name.
> Unnamed extensions are visible only
> in the library where they're declared.
> Since they don't have a name,
> there is no way to explicitly apply them
> to resolve [API conflicts](#api-conflicts).

Reviewer: **cleanup:** This still needs active voice to clean it up.
```suggestion
When declaring an extension, you can omit the name.
An unnamed extension can be used only in the library
where it's declared.
Since they don't have a name,
they cannot be explicitly applied
to resolve [API conflicts](#api-conflicts).
```

**note:** If you say _visible_, you need to say visible to _whom_.

Author: I'm rather hesitant to lose the word visible here, "visible inside" or "visible in" a library, package, module, etc. is a common construct in programming languages and their docs, including ours.

Reviewer: @parlough : What does "visible" mean? Visible to whom? What does it mean to not be visible? That's what I'm trying to understand.

Author: @atsansone By "visible in a library", I mean visible to other code (at least) within the same library, visible to the analyzer and compiler(s) when determining what is accessible in the library, in some-cases visible to reflective accessors of the library through `dart:mirrors`.

If it's visible only to that library (private), it won't show up in the `dart doc` output for that library, other libraries can't access the extension, the analyzer won't suggest it when coding in other libraries.

If it's not private, that means other libraries can import it making its members will be visible and accessible, `dart doc` can include the member in its rendered output, etc.

Essentially, I do think "can be used only in the library where it's declared" covers the primary use case of a developer coding, but it's also about whether its discoverable, included in various outputs, etc.
